MetaMask/metamask-extension
# Restore accounts with tokens

## **Description**

When onboarding an existing wallet, MetaMask currently restores accounts that have an ETH/native balance.  This PR will also restore accounts that have ERC20 token balances, even if they have no ETH.  

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/7811

## **Manual testing steps**

1. Setup a wallet where the first account has ETH and the second account has 0 ETH but does have ERC20 token(s).
2. Import the seed into MetaMask
3. The first 2 accounts should be restored

## **Screenshots/Recordings**

### **Before**

https://github.com/MetaMask/metamask-extension/assets/3500406/2dc7be22-3155-409b-8903-28f30cf1bc21

### **After**

https://github.com/MetaMask/metamask-extension/assets/3500406/b963137d-65d8-4a34-ae32-477996fa276c

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
